cdk/lib/kafkaEventBridgeSinkStack.ts

```diff
@@ -0,0 +1,203 @@
+            vpc,
+            ecsCluster: producer.ecsCluster,
+            bootstrapServers: bootstrapServers.getResponseField('BootstrapBrokerStringSaslIam'),
+            region: this.region,
+            account: this.account,
+            clusterName: cluster.clusterName,
+            schemaRegistry: schemaRegistry,
+        })
+
+        transactionAnalyzer.node.addDependency(producer)
+
+        mskSG.addIngressRule(producer.securityGroup, Port.tcp(9098))
+        mskSG.addIngressRule(transactionAnalyzer.securityGroup, Port.tcp(9098))
+        mskSG.addIngressRule(mskSG, ec2.Port.tcp(9098))
+        mskSG.addIngressRule(Peer.ipv4(vpc.vpcCidrBlock), Port.tcp(9098))
+
+        const connectorLogGroup = new logs.LogGroup(this, 'connectorLogGroup', {
+            logGroupName: '/aws/mskconnect/eventBridgeSinkConnector',
+            removalPolicy: RemovalPolicy.DESTROY,
+            retention: RetentionDays.ONE_WEEK
+        })
+
+        const connectorRole = new iam.Role(this, 'connectorRole', {
+            assumedBy: new iam.ServicePrincipal('kafkaconnect.amazonaws.com')
+        })
+
+        const connectorSG = new ec2.SecurityGroup(this, 'connectorSG', {
+            securityGroupName: `mskConnectSecurityGroup-${props.stackName}`,
+            description: 'Security group for Amazon MSK Connect',
+            vpc
+        })
+
+
+
+        if (props.deploymentMode === 'FULL') {
+            const connector = new Connector(this, 'connector', {
+                vpc,
+                bootstrapServers: bootstrapServers.getResponseField('BootstrapBrokerStringSaslIam'),
+                region: this.region,
+                account: this.account,
+                clusterName: cluster.clusterName,
+                stackName: this.stackName,
+                connectorSG,
+                connectorRole,
+                connectorLogGroup
+            })
+        }
+
+        connectorRole.addToPolicy(new iam.PolicyStatement({
+            actions: ['events:PutEvents'],
+            resources: [`arn:aws:events:${this.region}:${this.account}:event-bus/eventbridge-sink-eventbus`]
+        }))
+
+        connectorRole.addToPolicy(new iam.PolicyStatement({
+            actions: ['glue:GetSchemaVersion'],
+            resources: [`*`]
+        }))
+
+        connectorRole.addToPolicy(new PolicyStatement({
+            actions: [
+                "kafka-cluster:Connect",
+                "kafka-cluster:AlterCluster",
+                "kafka-cluster:DescribeCluster"
+            ],
+            resources: [`arn:aws:kafka:${this.region}:${this.account}:cluster/${cluster.clusterName}/*`]
+        }))
+
+        connectorRole.addToPolicy(new PolicyStatement({
+            actions: [
+                "kafka-cluster:*Topic*",
+                "kafka-cluster:WriteData",
+                "kafka-cluster:ReadData"
+            ],
+            resources: [`arn:aws:kafka:${this.region}:${this.account}:topic/${cluster.clusterName}/*`]
```

Comment:
nit: if topic name would be a property used throughout this demo (producer, consumer, IAM), this could be scoped down further to prevent broad IAM access